soccer/gameplay/tactics/advance_zone_midfielder.py

```diff
@@ -0,0 +1,138 @@
+        self.names = ['left_mid', 'right_mid']
+        # Which robot would be best to move to first move point
+        self.optimal = 0
+
+        self.moves = [None, None]
+
+        for s in AdvanceZoneMidfielder.State:
+            self.add_state(s, behavior.Behavior.State.running)
+
+        # more conservative play that should be implemented later
+        self.add_transition(
+            AdvanceZoneMidfielder.State.passSet,
+            AdvanceZoneMidfielder.State.passBack,
+            lambda: False,  #not self.should_get_open(),
+            "When dribbler is in danger and not about to shoot")
+
+        #intially should be passSet method
+        self.add_transition(behavior.Behavior.State.start,
+                            AdvanceZoneMidfielder.State.passSet, lambda: True,
+                            "Immediately")
+
+    def execute_passBack(self):
+        target_y = self.y_hold_percent * main.ball().pos.y
+
+        for i in range(2):
+            if (self.moves[i] is None):
+                self.moves[i] = skills.move.Move(
+                    robocup.Point(self.x_positions[i], target_y))
+                self.add_subbehavior(
+                    self.moves[i], self.names[i], required=False, priority=1)
+            else:
+                self.moves[i].pos = robocup.Point(self.x_positions[i],
+                                                  target_y)
+
+    def execute_passSet(self):
+        #gets the best position to travel to for ball reception
+        best_Pos_From_Ball, value = evaluation.passing_positioning.eval_best_receive_point(
+            main.ball().pos,
+            main.our_robots(), AdvanceZoneMidfielder.FIELD_POS_WEIGHTS,
+            AdvanceZoneMidfielder.NELDER_MEAD_ARGS,
+            AdvanceZoneMidfielder.PASSING_WEIGHTS)
+
+        points = [robocup.Point(0, 0), robocup.Point(0, 0)]
+        self.optimal = 0
+        # sets either the left or the right side of the ball to the best ball position
+        # and then notes which one is the other
+        # left side of the ball
+        if (best_Pos_From_Ball.x < 0):
+            points[0] = best_Pos_From_Ball
+            self.optimal = 0
+            other = 1
+        #right side of the ball
+        else:
+            points[1] = best_Pos_From_Ball
+            self.optimal = 1
+            other = 0
+
+        # sets the second point
+        points[
+            other], value2 = evaluation.passing_positioning.eval_best_receive_point(
+                points[self.optimal],
+                main.our_robots(), AdvanceZoneMidfielder.FIELD_POS_WEIGHTS,
+                AdvanceZoneMidfielder.NELDER_MEAD_ARGS,
+                AdvanceZoneMidfielder.PASSING_WEIGHTS)
+
+        # moves the robots
+        for i in range(2):
+            if (self.moves[i] is None):
+                self.moves[i] = skills.move.Move(points[i])
+                self.add_subbehavior(
+                    self.moves[i], self.names[i], required=False, priority=1)
```

Comment:
I'm not sure if this is needed, but if we do that simplification from above, we can set the optimal point to have a higher priority than the second point